### PR TITLE
Provide an Interleaved Tables Sample

### LINKED ISSUE #196: Create example with @Interleaved
Sample of how to use interleaved entity would be helpful. 
For instance combination of @GeneratedValue, and @Interleaved when creating entries is not obvious

### PR DESCRIPTION
**Pull request by dzou**
Provide an `@Interleaved` sample demonstrating how to setup the entities.

Fixes #196 .

### CODE REVIEW

File: google-cloud-spanner-hibernate-samples/basic-hibernate-sample/src/main/java/com/example/SampleApplication.java [hunk 1]
```diff
@@ -49,6 +55,35 @@ public static void main(String[] args) {
...
+    singer.addAlbum(album);
+
+    session.save(singer);
+    session.save(album);
+    session.getTransaction().commit();
+
+    List<Singer> singers =
+        session.createQuery("from Singer", Singer.class).list();
+    System.out.println(
+        String.format("There are %d singer saved in the table:", singers.size()));
```
Comment (meltsufin) [inline google-cloud-spanner-hibernate-samples/basic-hibernate-sample/src/main/java/com/example/SampleApplication.java:82]:
```suggestion
        String.format("There are %d singers saved in the table:", singers.size()));
```

File: google-cloud-spanner-hibernate-samples/basic-hibernate-sample/src/main/java/com/example/entities/Album.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  @Type(type = "uuid-char")
+  private UUID albumId;
+
+  @Id
+  @ManyToOne
+  @JoinColumn(name = "singerId")
+  @Type(type = "uuid-char")
+  private Singer singer;
+
+  String title;
```
Comment (meltsufin) [inline google-cloud-spanner-hibernate-samples/basic-hibernate-sample/src/main/java/com/example/entities/Album.java:51]: For consistency:
```suggestion
  private String title;
```

File: google-cloud-spanner-hibernate-samples/basic-hibernate-sample/src/main/java/com/example/entities/Album.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+  public Album(Singer singer, String title) {
+    this.singer = singer;
+    this.title = title;
+  }
+
+  Album() {
```
Comment (meltsufin) [inline google-cloud-spanner-hibernate-samples/basic-hibernate-sample/src/main/java/com/example/entities/Album.java:58]:
```suggestion
  public Album() {
```

File: google-cloud-spanner-hibernate-samples/basic-hibernate-sample/src/main/java/com/example/SampleApplication.java [hunk 4]
```diff
@@ -49,6 +55,35 @@ public static void main(String[] args) {
 
     // Save an entity into Spanner Table.
```
Comment (meltsufin) [inline google-cloud-spanner-hibernate-samples/basic-hibernate-sample/src/main/java/com/example/SampleApplication.java:56]:
```suggestion
    // Save a Person entity into Spanner Table.
```

**Comment by dzou** [comment]
>It looks like a lot more is going on now in this one sample.
Would it make sense to create a 2nd sample, or would it add too much maintenance?

I gave trying to create a new sample a shot, but it didn't seem worth it because there was too much duplication of config/setup code.

I ended up just leaving keeping it 1 sample and adding some more description to the sample README, hope this is sufficiently clear.

**Comment by dzou** [comment]
> Sorry for wasting your time with the suggestion to split the sample.
> This looks good to me.

No problem!

**Comment by dzou** [comment]
Could you reapprove again? I submitted/reverted a dummy change because the cla/google check got stuck.

**Comment by dzou** [comment]
Thanks!

**Pull request merged**